md.py

```diff
@@ -235,21 +235,29 @@ def stop(self):
         while pathlib.Path(self.md_dev).exists():
             time.sleep(0.01)
 
-    def _create_loop_disk(self, i, size):
+    def _get_loop_paths(self, i):
         dev = f"/dev/loop{i}"
         backing = f"/var/tmp/lodisk{i}"
+        return dev, backing
+
+    def _create_loop_disk(self, dev, backing, size):
         subprocess.call(["losetup", "-d", dev], stdout=subprocess.DEVNULL,
                         stderr=subprocess.DEVNULL)
         open(backing, "w").close()
         os.truncate(backing, size)
         subprocess.check_call(["losetup", dev, backing])
-        return dev
 
     def _create_loop_disks(self, count, size):
         ret = []
+        off = 0
 
         for i in range(count):
-            dev = self._create_loop_disk(i, size)
+            dev, backing = self._get_loop_paths(i + off)
+            while pathlib.Path(dev).exists() and \
+                  not pathlib.Path(backing).exists():
+                off += 1
+                dev, backing = self._get_loop_paths(i + off)
+            self._create_loop_disk(dev, backing, size)
```

Comment:
I think I'd prefer to just stop if there are loop devices already configured and tell the user. No sense trying to work around that and create other weird issues.

Reply:
Why? I don't know or care what else is creating a loop dev in my vm, but it should still be able to run the tests

Reply:
You should know or care what else is creating a loop dev. What happens if one is left over due to a bug? Just ignore it and keep running until you run out? Don't run on a system you don't understand.

Reply:
The code checks to see if the loop dev is ours via the backing files we use (which are never removed) it's external applications I don't care about.

Reply:
I still think it's better for reproducibility. Less confusing. The mdadm tests do the same thing: refuse to run if there are any md devices configured and requires there to be no loop devices if that method was selected. 

This is meant for test systems with relatively controlled environments. Not systems your running with other loop devices doing other things.